android/engine/src/main/java/org/smartregister/fhircore/engine/rulesengine/RulesFactory.kt

```diff
@@ -150,30 +149,27 @@ constructor(
     }
   }
 
-  fun generateRules(ruleConfigsKey: String, ruleConfigs: List<RuleConfig>): Rules {
-    val jexlRules =
-      ruleConfigsCache.getOrDefault(
-        ruleConfigsKey,
-        Rules(
-            runBlocking(Dispatchers.Default) {
-                ruleConfigs.map { ruleConfig ->
-                  val customRule: JexlRule =
-                    JexlRule(jexlEngine)
-                      .name(ruleConfig.name)
-                      .description(ruleConfig.description)
-                      .priority(ruleConfig.priority)
-                      .`when`(ruleConfig.condition.ifEmpty { TRUE })
-
-                  ruleConfig.actions.forEach { customRule.then(it) }
-                  customRule
-                }
-              }
-              .toSet()
-          )
-          .also { ruleConfigsCache[ruleConfigsKey] = it }
-      )
-
-    return jexlRules
+  suspend fun generateRules(ruleConfigs: List<RuleConfig>): Rules {
+    val start = System.currentTimeMillis()
```

Comment:
You can use Kotlin's lambda function [measureTimeMillis](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.system/measure-time-millis.html)